netket/sampler/rules/custom_numpy.py

```diff
@@ -92,7 +91,7 @@ def transition(rule, sampler, machine, parameters, state, rng, σ):
 @jit(nopython=True)
 def _pick_random_and_init(batch_size, move_cumulative, out):
     for i in range(batch_size):
-        p = _random.uniform()
+        p = np.random.uniform(0.0, 1.0)
```

Comment:
sorry just so I remember, does this random have different seeds on different mpi ranks?

Reply:
my bad.
Numba does not support the numpy random number generators yet, so we have to generate those numbers outside the jit block.

The generator keeps the seed on every rank

Reply:
The rngs are seeded per-rank here https://github.com/netket/netket/blob/a48d23e7ece03cd46fbd61b10b363d59bb816aeb/netket/sampler/base.py#L211